framework/src/loops/ComputeElemAuxVarsThread.C

```diff
@@ -112,8 +115,27 @@ ComputeElemAuxVarsThread<AuxKernelType>::onElement(const Elem * elem)
       _fe_problem.reinitMaterials(elem->subdomain_id(), _tid);
 
     for (const auto & aux : kernels)
+    {
       aux->compute();
 
+      // update the aux solution vector if writable coupled variables are used
+      if (aux->hasWritableCoupledVariables())
+      {
+        Threads::spin_mutex::scoped_lock lock(writable_variable_mutex);
+        for (auto * var : aux->getWritableCoupledVariables())
+        {
+          // insert into the global solution vector
+          var->insert(_aux_sys.solution());
+          var->prepareAux();
```

Comment:
A UOthat is executed later might want to use the variables written to by an earlier UO. For the values at the QP to be updated wouldn't I need to re-prepare the element?

Reply:
If you are adding this line because of the name of the API, then that makes a lot of sense. The issue is that `prepareAux` hardly does any preparation at all. All it does is set `_has_dof_values = false` ... in a way `prepareAux` just marks our variable as unprepared ... what I suppose is preparation in a way 😆